src/main/java/io/cryostat/core/net/MemoryMetrics.java

```diff
@@ -0,0 +1,157 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.core.net;
+
+import java.util.Map;
+
+public class MemoryMetrics {
+    private final MemoryUsage heapMemoryUsage;
+    private final MemoryUsage nonHeapMemoryUsage;
+    private final long objectPendingFinalizationCount;
+    private final long freeHeapMemory;
+    private final long freeNonHeapMemory;
+    private final double heapMemoryUsagePercent;
+    private final boolean verbose;
+
+    public MemoryMetrics(Map<String, Object> attributes) {
+        this.heapMemoryUsage =
+                (MemoryUsage) attributes.getOrDefault("HeapMemoryUsage", new MemoryUsage());
+        this.nonHeapMemoryUsage =
+                (MemoryUsage) attributes.getOrDefault("NonHeapMemoryUsage", new MemoryUsage());
+        this.objectPendingFinalizationCount =
+                (int) attributes.getOrDefault("ObjectPendingFinalizationCount", Integer.MIN_VALUE);
+        this.freeHeapMemory = (long) attributes.getOrDefault("FreeHeapMemory", Long.MIN_VALUE);
+        this.freeNonHeapMemory =
+                (long) attributes.getOrDefault("FreeNonHeapMemory", Long.MIN_VALUE);
+        this.heapMemoryUsagePercent =
+                (double) attributes.getOrDefault("HeapMemoryUsagePercent", Double.MIN_VALUE);
+        this.verbose = (boolean) attributes.getOrDefault("Verbose", false);
+    }
+
+    public MemoryUsage getHeapMemoryUsage() {
+        return heapMemoryUsage;
+    }
+
+    public MemoryUsage getNonHeapMemoryUsage() {
+        return nonHeapMemoryUsage;
+    }
+
+    public long getObjectPendingFinalizationCount() {
+        return objectPendingFinalizationCount;
+    }
+
+    public long getFreeHeapMemory() {
+        return freeHeapMemory;
+    }
+
+    public long getFreeNonHeapMemory() {
+        return freeNonHeapMemory;
+    }
+
+    public double getHeapMemoryUsagePercent() {
+        return heapMemoryUsagePercent;
+    }
+
+    public boolean isVerbose() {
+        return verbose;
+    }
+
+    @Override
+    public String toString() {
+        return "MemoryMetrics{"
+                + ", heapMemoryUsage="
+                + heapMemoryUsage
+                + ", nonHeapMemoryUsage="
+                + nonHeapMemoryUsage
+                + ", objectPendingFinalizationCount="
+                + objectPendingFinalizationCount
+                + ", freeHeapMemory="
+                + freeHeapMemory
+                + ", freeNonHeapMemory="
+                + freeNonHeapMemory
+                + ", heapMemoryUsagePercent="
+                + heapMemoryUsagePercent
+                + ", verbose="
+                + verbose
+                + '}';
+    }
+    // Gson cannot read private fields of java.lang.management.MemoryUsage
```

Comment:
Rather than placing this wrapper into `-core`, which doesn't know anything about gson, I think this can be worked around in the application layer(s) that need to be able to serialize it. For the Cryostat backend that probably just means implementing a `PluggableTypeAdapter` that can serialize the `java.lang.management.MemoryUsage` with an explicit strategy so Gson doesn't try to use reflection for it. The deserialization operation can just be unsupported.

Reply:
Or, at worst, this serialization wrapper can go into the application layer. The Cryostat backend uses gson but the Agent uses Jackson, for example, so there will likely be differences in how this type can be serialized.

Reply:
Sure that can be done, there is just one thing, a default `MemoryUsage` object can't be created with custom values like Long.MIN_VALUE. Something like this can work though:
```
{
   init: -1,
   used: 0,
   commited: 0,
   max: -1,
}
```
Is that okay?

Reply:
I think that's fine. Those values match the JDK docs, which is probably what an end user would expect to see and where they would likely try to find references for the meanings of values:

https://docs.oracle.com/javase/7/docs/api/java/lang/management/MemoryUsage.html#getInit()